torchvision/csrc/io/image/cuda/encode_decode_jpegs_cuda.h

```diff
@@ -2,14 +2,15 @@
 
 #include <torch/types.h>
 #include "../image_read_mode.h"
+#include "decode_jpegs_cuda.h"
 #include "encode_jpegs_cuda.h"
 
 namespace vision {
 namespace image {
 
-C10_EXPORT torch::Tensor decode_jpeg_cuda(
-    const torch::Tensor& data,
-    ImageReadMode mode,
+C10_EXPORT std::vector<torch::Tensor> decode_jpegs_cuda(
+    const std::vector<torch::Tensor>& encoded_images,
```

Comment:
Why is this not a single torch::Tensor that's stacked as opposed to a std::vector of Tensors?

Reply:
Because the encoded jpegs are represented as variable length byte streams of type tensor(1). They cannot be stacked into a batch size they don't have the same dimensions.